.github/workflows/checks.yml

```diff
@@ -0,0 +1,56 @@
+name: Run checks and tests
+
+on:
+  - pull_request
+
+jobs:
+
+  checks:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.13"
+          cache: "pip"
+
+      - name: Install dependencies
+        run: |
+          pip install --upgrade pip
+          pip install ruff
+
+      - name: Run linter checks with ruff
+        run: ruff check
+
+      - name: Run formatter checks with ruff
+        run: ruff format --check
+
+  tests:
+    needs: checks
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        python-version: ["3.10", "3.11", "3.12", "3.13"]
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: "pip"
+
+      - name: Install dependencies
+        run: |
+          pip install --upgrade pip
+          pip install pytest coverage
+          pip install .
+
+      - name: Run tests and generate coverage report
+        run: |
+          coverage run -m pytest --run-marks-subset "fast" tests/
+          coverage html
+          coverage xml
+
+      - name: Upload coverage to codecov
+        uses: codecov/codecov-action@v5
```

Comment:
@wigging is there a reason to remove the condition that uploads the coverage for a single Python version?

```
      - name: Upload coverage report
        if: ${{ matrix.python-version == 3.12 }} # Only upload coverage once
        uses: codecov/codecov-action@v5
```

Reply:
No reason. I don't know why I removed it. Looks like you fixed it though. Thanks.